Webapp/sources/auxiliary.py

```diff
@@ -523,3 +524,23 @@ def smiles_to_inchi(smiles: str) -> Optional[str]:
 
 def remove_spaces_and_dashes(name: str) -> str:
     return name.replace(' ', '').replace('-', '')
+
+
+def abort_if_user_not_in_workbook(workgroup_name: str, workbook_name: str, workbook: models.WorkBook):
+    """
+    We abort the current process if the user is not in the workbook
+    Request values come from the frontend
+    """
+    if workbook is not None and not security_member_workgroup_workbook(workgroup_name, workbook_name):
+        abort(400)
```

Comment:
Should be a 401 unauthorised, as I think you're checking for permission here?

Reply:
Cheers Andy, I'll correct these and push changes